### PR TITLE
[Fix #1839] Remove Rainbow string monkey patching

### LINKED ISSUE #1839: Possible issue with calling .color
For some reason, gems that include rainbow, like RuboCop, are failing when used with Ruby 2.2.1. This might be a bug with Ruby's 2.2.x refinements feature.

```
private method `color' called for "An error occurred while Style/MultilineOperationIndentation":String
/Users/justin/.rvm/gems/ruby-2.2.1@madrone/gems/rubocop-0.30.1/lib/rubocop/cop/team.rb:93:in `block (2 levels) in process_commissioner_errors'
```

## UPDATE
Actually, I'm seeing this with Ruby 2.1.5. Researching.

### PR DESCRIPTION
**Pull request by daviddavis**

### CODE REVIEW

File: lib/rubocop/config.rb [hunk 1]
```diff
@@ -80,7 +80,7 @@ def cop_enabled?(cop)
     def warn_unless_valid
       validate
     rescue Config::ValidationError => e
-      warn "Warning: #{e.message}".color(:red)
+      warn Rainbow.new.wrap("Warning: #{e.message}").red
```
Comment (daviddavis) [inline lib/rubocop/config.rb]: Need to figure out why I keep getting `undefined method `Rainbow' for module `Kernel'` here when I try to call `Rainbow("Warning: #{e.message}")`,

**Comment by bbatsov** [comment]
In the commit message - "Fixes" -> "Fix". This also needs a changelog entry.

File: lib/rubocop/config.rb (hunk 1 above)
Comment (rrosenblum) [inline lib/rubocop/config.rb]: What is the difference between `Rainbow("some string").red` and `Rainbow.new.wrap("some string").red`?

Comment (daviddavis) [inline lib/rubocop/config.rb]: Basically, nothing. Just one is a global function that calls #wrap:

http://git.io/vGHUg

**Comment by bbatsov** [comment]
:+1: Thanks. Hopefully, later we'll solve the mystery regarding the `Rainbow` function.

**Pull request merged**